File: util/timeline_adjust.py
#!/usr/bin/python
from __future__ import print_function
import argparse
import re

time_re = re.compile(r"^\s*#?\s*([0-9]+(?:\.[0-9]+)?)\s+\"")
first_num_re = re.compile(r"([0-9]+(?:\.[0-9]+)?)")


def adjust_lines(lines, adjust):
  for line in lines:
    match = re.match(time_re, line)
    if match:
      time = float(match.group(1)) + adjust
      print(re.sub(first_num_re, str(time), line, 1), end='')
    else:
      print(line, end='')

def main():
  parser = argparse.ArgumentParser(
      description="A utility to uniformly adjust times in an act timeline file")
  parser.add_argument('--file', required=True, type=argparse.FileType('r', encoding="utf8"),
      help="The timeline file to adjust times in")
  parser.add_argument('--adjust', required=True, type=float,
      help="The amount of time to adjust each entry by")

  args = parser.parse_args()
  adjust_lines(args.file, args.adjust)


if __name__ == "__main__":
  main()

File: util/timeline_adjust.mjs
import fs from 'fs';
import { ArgumentParser } from 'argparse';

const timeRe = /^\s*#?\s*([0-9]+(?:\.[0-9]+)?)\s+"/;
const firstNumRe = /([0-9]+(?:\.[0-9]+)?)/;

const adjustLines = (lines, adjust) => {
  for (const line of lines) {
    const match = timeRe.exec(line);
    if (match) {
      const time = parseFloat(match[1]) + adjust;
      console.log(line.replace(firstNumRe, time.toFixed(1)));
    } else {
      console.log(line);
    }
  }
};

const main = () => {
  const parser = new ArgumentParser({ description: 'Argparse example' });

  parser.addArgument(['-f', '--file'], {
    help: 'The timeline file to adjust times in',
  });
  parser.addArgument(['-a', '--adjust'], {
    type: Number,
    help: 'The amount of time to adjust each entry by',
  });

  const args = parser.parseArgs();
  adjustLines(fs.readFileSync(args.file).toString().split('\n'), args.adjust);
};

void main();
